packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -865,47 +861,105 @@ export class ChannelCollection implements IFluidDataStoreChannel, IDisposable {
+	 * @param messageContents - The contents, local metadata and clientSequenceNumbers of the messages.
+	 * @param local - true if the messages were originally generated by the client receiving it.
+	 */
+	private processChannelMessages(
+		message: ISequencedRuntimeMessageCore,
+		messageContents: IRuntimeMessageContents[],
 		local: boolean,
-		localMessageMetadata: unknown,
-	) {
-		const context = this.contexts.get(address);
+	): void {
+		let currentMessageState: { address: string; type: string } | undefined;
+		let currentMessageContents: IRuntimeMessageContents[] = [];
+
+		// Helper that sends the current bunch of messages to the data store. It validates that the data stores exists.
+		const sendBunchedMessages = () => {
+			assert(currentMessageState !== undefined, "message state must exist to send messages");
+			const currentContext = this.contexts.get(currentMessageState.address);
+			assert(!!currentContext, "Context not found");
+
+			currentContext.processMessages(
+				{ ...message, type: currentMessageState.type },
+				currentMessageContents,
+				local,
+			);
+			currentMessageContents = [];
+		};
 
-		// If the data store has been deleted, log an error and ignore this message. This helps prevent document
-		// corruption in case a deleted data store accidentally submitted an op.
-		if (this.checkAndLogIfDeleted(address, context, "Changed", "processFluidDataStoreOp")) {
-			return;
-		}
+		/**
+		 * Bunch contiguous messages for the same data store and send them together.
+		 * This is an optimization mainly for DDSes, where it can process a bunch of ops together. DDSes
+		 * like merge tree or shared tree can process ops more efficiently when they are bunched together.
+		 */
+		for (const { contents, ...restOfMessageContents } of messageContents) {
+			const envelope = contents as IEnvelope;
+			const address = envelope.address;
+			const context = this.contexts.get(address);
+
+			// If the data store has been deleted, log an error and ignore this message. This helps prevent document
+			// corruption in case a deleted data store accidentally submitted an op.
+			if (this.checkAndLogIfDeleted(address, context, "Changed", "processFluidDataStoreOp")) {
+				continue;
+			}
 
-		if (context === undefined) {
-			// Former assert 0x162
-			throw DataProcessingError.create(
-				"No context for op",
-				"processFluidDataStoreOp",
-				message,
-				{
-					local,
-					messageDetails: JSON.stringify({
-						type: message.type,
-						contentType: typeof message.contents,
-					}),
-					...tagCodeArtifacts({ address }),
-				},
+			if (context === undefined) {
+				// Former assert 0x162
+				throw DataProcessingError.create(
+					"No context for op",
+					"processFluidDataStoreOp",
+					message as ISequencedDocumentMessage,
+					{
+						local,
+						messageDetails: JSON.stringify({
+							type: message.type,
+							contentType: typeof contents,
+						}),
+						...tagCodeArtifacts({ address }),
+					},
+				);
```

Comment:
There is another similar throw in this file. Is there a chance to share more code (i.e. have a function that is something like throwOnUnknownContext(message)?

Reply:
The other throws are different in some ways so creating a helper doesn't seem very beneficial.